control-plane-operator/controllers/hostedcontrolplane/clusterpolicy/deployment_test.go

```diff
@@ -0,0 +1,48 @@
+package clusterpolicy
+
+import (
+	hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
+	"github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/manifests"
+	"github.com/openshift/hypershift/support/config"
+	"github.com/openshift/hypershift/support/util"
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+	"testing"
+)
+
+func TestReconcileClusterPolicyDeployment(t *testing.T) {
+
+	imageName := "clusterPolicyImage"
+	// Setup expected values that are universal
+
+	// Setup hypershift hosted control plane.
+	targetNamespace := "test"
+	clusterPolicyDeployment := manifests.ClusterPolicyControllerDeployment(targetNamespace)
+	hcp := &hyperv1.HostedControlPlane{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "hcp",
+			Namespace: targetNamespace,
+		},
+	}
+	hcp.Name = "name"
+	hcp.Namespace = "namespace"
+	ownerRef := config.OwnerRefFrom(hcp)
+
+	testCases := []struct {
+		deploymentConfig config.DeploymentConfig
+	}{
+		// empty deployment config
+		{
+			deploymentConfig: config.DeploymentConfig{},
+		},
+	}
+	for _, tc := range testCases {
+		expectedTermGraceSeconds := clusterPolicyDeployment.Spec.Template.Spec.TerminationGracePeriodSeconds
+		expectedMinReadySeconds := clusterPolicyDeployment.Spec.MinReadySeconds
+		err := ReconcileDeployment(clusterPolicyDeployment, ownerRef, imageName, tc.deploymentConfig, util.AvailabilityProberImageName, pointer.Int32(1234))
+		assert.NoError(t, err)
+		assert.Equal(t, expectedTermGraceSeconds, clusterPolicyDeployment.Spec.Template.Spec.TerminationGracePeriodSeconds)
```

Comment:
I think we can just make an assertion that `clusterPolicyDeployment.Spec.Template.Spec.TerminationGracePeriodSeconds` is empty value or not. We probably don't want to compare it against what's being set from `manifests.ClusterPolicyControllerDeployment(targetNamespace)`, since that could theoretically change later on. And at that point, this unit test will not catch that.